e2etests/test/server_valid.js

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var serviceSdk = require('azure-iothub');
+var serviceSas = require('azure-iothub').SharedAccessSignature;
+var provisioningServiceClient = require('azure-iot-provisioning-service').ProvisioningServiceClient;
+var Message = require('azure-iot-common').Message;
+var anHourFromNow = require('azure-iot-common').anHourFromNow;
+var Registry = require('azure-iothub').Registry;
+var NoRetry = require('azure-iot-common').NoRetry;
+var deviceSdk = require('azure-iot-device');
+
+
+var uuid = require('uuid');
+
+var hubConnectionString = process.env.IOTHUB_CONN_STRING_INVALIDCERT;
+var deviceConnectionString = process.env.IOTHUB_DEVICE_CONN_STRING_INVALIDCERT;
+var dpsDeviceClientEndpoint = process.env.DPS_GLOBALDEVICEENDPOINT_INVALIDCERT;
+
+
+var correctDisconnectMessage = function(err, done) {
+  if (err) {
+    if (err.amqpError && (err.amqpError.name === 'NotConnectedError')) {
+      done();
+    } else if (err.name && (err.name  === 'NotConnectedError')) {
+      done();
+    } else if (err.code && (err.code === 'UNABLE_TO_VERIFY_LEAF_SIGNATURE')) {
+      done();
+    } else {
+      done(new Error('client did NOT detect bad cert.'));
```

Comment:
not true. it looks like we did detect it and it's just not the error you're expecting.

Reply:
Now you're nitting.  Something happened.  The most likely scenario is that we had a name mismatch on a link.  It was not the tls error that we were looking for.

---
In reply to: [220400746](https://github.com/Azure/azure-iot-sdk-node/pull/378#discussion_r220400746) [](ancestors = 220400746)